versioned_docs/version-2021sp/lecture10.md

```diff
@@ -3,4 +3,101 @@ id: lecture10
 title: Lecture 10
 ---
 
-Coming soon!
+[Lecture Slides](https://docs.google.com/presentation/d/1_DbVZInabluIm2QDOkgWZG0SwecCYTcyOInm_PIQELE/edit?usp=sharing)
+
+[Lecture Video](https://drive.google.com/file/d/1d2mCrRM_mavlNhCJk0kf-6ZoTS9VvVGP/view?usp=sharing)
+
+[Final Project - Milestone 2 due 5/6 3:59 PM](/docs/finalproject#milestone-2)
+
+## React Native
+
+### What is React Native?
+
+React Native allows for cross platform mobile development using a webdev framework we already know--React!
+
+React Native allows you to build UIs independent of the platform. Usually when developing an app you have to develop an Android version (using Java/Kotlin) and iOS version (using Objective-C/Swift) separately. React Native takes care of this conversion for you.
+
+### Core React Native Components
+
+Since React Native is really just React, many of the same concepts (`useState`, `props`, React Hooks, etc) still apply to React. However, instead of HTML we have `Views`. A view is the basic building block of UI in mobile development. Views can display images, hold text, handle user input, etc.
+
+Some core React Native components are:
+
+- `<View>`: A container that supports layout with flexbox, style, some touch handling, and accessibility controls. Similar to a non-scrolling `<div>`.
+- `<Text>`: Displays, styles, and nests strings of text and even handles touch events. Similar to a `<p>`
+- `<Image>`: Displays images like `<img>`
+- `<ScrollView>`: A generic scrolling container than can hold nested components and views. Similar to a `<div>`.
+- `<TextInput>`: User text input field similar to `<input type="text" />`.
+- ... and you can also define your own custom components (and use those built by the community)!
+
+### How to start a React Native Project?
+
+React Native uses `Expo` framework to develop, build, and iterate on iOS, Android and webapps. Expo provides a UI for you to view your changes and if you download the Expo app ([Android](https://play.google.com/store/apps/details?id=host.exp.exponent&hl=en_US), [iOS](https://apps.apple.com/us/app/expo-client/id982107779)) you can see those changes on your phone as well! After all, we're doing mobile development.
```

Comment:
This might be a bit misleading, you can use React Native without Expo

Reply:
Okay, i'll note that in the notes